tests/conftest.py

```diff
@@ -364,10 +367,23 @@ def _common_wheel_editable_install(
     wheel_candidates = list(common_wheels.glob(f"{package}-*.whl"))
     assert len(wheel_candidates) == 1, wheel_candidates
     install_dir = tmpdir_factory.mktemp(package) / "install"
-    Wheel(wheel_candidates[0]).install_as_egg(install_dir)
-    (install_dir / "EGG-INFO").rename(install_dir / f"{package}.egg-info")
-    assert compileall.compile_dir(str(install_dir), quiet=1)
-    return install_dir
+    lib_install_dir = install_dir / "lib"
+    bin_install_dir = install_dir / "bin"
+    with zipfile.ZipFile(wheel_candidates[0]) as zf:
+        install(
+            WheelFile(zf),
+            SchemeDictionaryDestination(
+                {
+                    "purelib": str(lib_install_dir),
+                    "platlib": str(lib_install_dir),
+                    "scripts": str(bin_install_dir),
+                },
+                interpreter=sys.executable,
```

Comment:
This would mean that the executables run with the same environment as the test runner, which is likely not what we want.

This should really be using the Python executable from the virtualenv. I think the right thing to do here would be to change up the `virtualenv` fixture, with the object coming from that getting a new `.install_from_wheel(path)` method.

Reply:
I think these scripts are never used. I'll remove the `bin` dir and see if the tests pass.
Then we can do a refactoring in followup, after #11288.

Reply:
Yea, let's do that!

I might end up picking up the refactoring, as part of https://github.com/pradyunsg/pip/tree/improve-test-isolation. :)